R/ignore.R

```diff
@@ -25,5 +25,7 @@ use_build_ignore <- function(files, escape = TRUE, base_path = ".") {
 escape_path <- function(x) {
   x <- gsub("\\.", "\\\\.", x)
   x <- gsub("/$", "", x)
+  # maintain absolute path on windows to keep double \\ even after escaping
+  x <- gsub("(^\\D:)", "\\1\\\\", x)
```

Comment:
this was a hacky? way of getting around the original gsub problems, discussed in the pkgdown issue. Basically on windows, this escape_path function previously would escape in such a way that the resultant path would be like

`C:\Users\hadley\...`

this causes the PCRE regex compilation error, which wants

`C:\\Users\hadley\...`

thus this regex will replace, if it detects a windows absolute path delineated by `<Driveletter>:\` and replace it with `<Driveletter>:\\` such that the regex doesn't explode.